catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/CqlCommands.java

```diff
@@ -0,0 +1,238 @@
+
+        if (StringUtils.isNotBlank(startDate) && StringUtils.isNotBlank(endDate)) {
+            return filterBuilder.attribute(temporalPropertyToFilter)
+                    .is()
+                    .during()
+                    .dates(formatter.parse(startDate), formatter.parse(endDate));
+        } else if (start > 0 && end > 0) {
+            return filterBuilder.attribute(temporalPropertyToFilter)
+                    .is()
+                    .during()
+                    .dates(new Date(start), new Date(end));
+        } else if (isUseTemporal) {
+            return filterBuilder.attribute(temporalPropertyToFilter)
+                    .is()
+                    .during()
+                    .last(start);
+        } else {
+            if (caseSensitive) {
+                return filterBuilder.attribute(Metacard.ANY_TEXT)
+                        .is()
+                        .like()
+                        .caseSensitiveText(searchPhrase);
+            } else {
+                return filterBuilder.attribute(Metacard.ANY_TEXT)
+                        .is()
+                        .like()
+                        .text(searchPhrase);
+            }
+        }
+    }
+
+    protected long getFilterStartTime() {
+        if (lastSeconds > 0) {
+            return filterCurrentTime - TimeUnit.SECONDS.toMillis(lastSeconds);
+        } else if (lastMinutes > 0) {
+            return filterCurrentTime - TimeUnit.MINUTES.toMillis(lastMinutes);
+        } else if (lastHours > 0) {
+            return filterCurrentTime - TimeUnit.HOURS.toMillis(lastHours);
+        } else if (lastDays > 0) {
+            return filterCurrentTime - TimeUnit.DAYS.toMillis(lastDays);
+        } else if (lastWeeks > 0) {
+            Calendar weeks = GregorianCalendar.getInstance();
+            weeks.setTimeInMillis(filterCurrentTime);
+            weeks.add(Calendar.WEEK_OF_YEAR, -1 * lastWeeks);
+            return weeks.getTimeInMillis();
+        } else if (lastMonths > 0) {
+            Calendar months = GregorianCalendar.getInstance();
+            months.setTimeInMillis(filterCurrentTime);
+            months.add(Calendar.MONTH, -1 * lastMonths);
+            return months.getTimeInMillis();
+        } else {
+            return 0;
+        }
+    }
+
+    protected String getTemporalProperty() {
+        if (metacardTypes != null && StringUtils.isNotEmpty(temporalProperty)) {
+            return metacardTypes.stream()
+                    .map(metacardType -> metacardType.getAttributeDescriptors())
+                    .flatMap(Set::stream)
+                    .filter(isDateType())
+                    .map(AttributeDescriptor::getName)
+                    .filter(temporalProperty::equalsIgnoreCase)
+                    .findFirst()
+                    .orElse(DEFAULT_TEMPORAL_PROPERTY);
+        }
+
+        return DEFAULT_TEMPORAL_PROPERTY;
+    }
+
+    private Predicate<AttributeDescriptor> isDateType() {
```

Comment:
As an alternative, you could have a method that takes an AttributeDescriptor and returns boolean. The you would use it as: .filter(this::isDateType).